### ISSUE TITLE
AdamWOptimizer's first step is 3.16x the learning rate

### ISSUE THREAD

**Issue by Eric-Louchez** [Shorokoo/Shorokoo#205]
`AdamWOptimizer` omits bias correction. `training.md` records this and calls the consequence
"early-step behavior differs slightly from reference AdamW". It is not slight — the **first step is
3.16× the learning rate**.

Repro — one trainable scalar, one `TrainStep`, lr 0.1, β1 0.9, β2 0.999, ε 1e-8, weight decay 0:

```
AdamOptimizer : p 0.000000 -> -0.100000   |delta| = 0.100000
AdamWOptimizer: p 0.000000 -> -0.316230   |delta| = 0.316230
```

`AdamOptimizer` gives exactly `lr`, which is the correct bias-corrected first step. `AdamWOptimizer`
gives `lr·(1-β1)/sqrt(1-β2) = 0.1 · 0.1 / 0.0316 = 0.3162`.

With the default β2 = 0.999 the overshoot is 3.16×; it grows as β2 → 1 (β2 = 0.9999 gives 10×). At
weight decay 0, `AdamW` and `Adam` should be identical, and here they differ by a factor of three on
the step that matters most for stability.

The gap closes over the first few hundred steps, so it will not show up as an obvious failure —
just a worse early trajectory, and a warmup schedule that has to absorb it. A user who follows a
recipe saying "AdamW" and types `AdamWOptimizer` gets this silently.

Suggest either adding the timestep and bias correction, or renaming so the difference is visible at
the call site.
